x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/services/openai/OpenAiServiceSettings.java

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.inference.services.openai;
+
+import org.elasticsearch.TransportVersion;
+import org.elasticsearch.TransportVersions;
+import org.elasticsearch.common.ValidationException;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.core.Nullable;
+import org.elasticsearch.inference.ModelConfigurations;
+import org.elasticsearch.inference.ServiceSettings;
+import org.elasticsearch.xcontent.XContentBuilder;
+import org.elasticsearch.xpack.inference.services.MapParsingUtils;
+
+import java.io.IOException;
+import java.net.URI;
+import java.util.Map;
+
+import static org.elasticsearch.xpack.inference.services.MapParsingUtils.convertToUri;
+import static org.elasticsearch.xpack.inference.services.MapParsingUtils.createUri;
+
+/**
+ * Defines the base settings for interacting with OpenAI.
+ * @param uri an optional uri to override the openai url. This should only be used for testing.
+ */
+public record OpenAiServiceSettings(@Nullable URI uri) implements ServiceSettings {
```

Comment:
I don't anticipate there being any service settings for openai so I think we can get away with a single class for all openai operations.

The uri here is to allow unit testing to hit a mock server. Users won't need to set this in production, but maybe it'll be helpful for support proxies 🤷‍♂️ .